File: JS/Tema2_Ejercicio5.js
/*
 * TEMA 2: EXPRESIONES Y OPERADORES - EJERCICIO 5
 * ===============================================
 * Utilizar operador ternario para evaluaciones condicionales.
 * 
 * EXPLICACIÓN:
 * - Operador ternario: condición ? valorSiVerdadero : valorSiFalso
 * - Equivalente a un if-else en una sola expresión
 * - Útil para asignaciones condicionales simples
 */

const edad = 25;
const esEstudiante = true;
const dinero = 100;

// Operador ternario simple
const mayoria = edad >= 18 ? "Mayor de edad" : "Menor de edad";

// Operador ternario anidado
const descuento = esEstudiante ? 
    dinero * 0.15 : 
    dinero * 0.05;

const preciofinal = dinero - descuento;

console.log("=== OPERADOR TERNARIO ===");
console.log(`Edad: ${edad}`);
console.log(`Estado: ${mayoria}`);

console.log(`\nEs estudiante: ${esEstudiante}`);
console.log(`Dinero inicial: $${dinero}`);
console.log(`Descuento aplicado: $${descuento.toFixed(2)}`);
console.log(`Precio final: $${preciofinal.toFixed(2)}`);

// Clasificar números
const numero = 42;
const clasificacion = numero % 2 === 0 ? "PAR" : "IMPAR";
console.log(`\n${numero} es: ${clasificacion}`);

/*
 * SALIDA ESPERADA:
 * === OPERADOR TERNARIO ===
 * Edad: 25
 * Estado: Mayor de edad
 * 
 * Es estudiante: true
 * Dinero inicial: $100
 * Descuento aplicado: $15.00
 * Precio final: $85.00
 * 
 * 42 es: PAR
 */

File: Java/Tema2_Ejercicio5.java
/*
 * TEMA 2: EXPRESIONES Y OPERADORES - EJERCICIO 5
 * ================================================
 * Escriba un programa que demuestre el operador ternario (? :).
 * Este operador es una forma compacta de escribir un if-else.
 * 
 * EXPLICACIÓN:
 * - Sintaxis: condición ? valor_si_verdadero : valor_si_falso
 * - Si la condición es verdadera, retorna el primer valor
 * - Si la condición es falsa, retorna el segundo valor
 */

import java.util.Scanner;

public class Tema2_Ejercicio5 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite un número: ");
        int numero = scanner.nextInt();
        
        // Usar operador ternario
        String tipo = (numero % 2 == 0) ? "par" : "impar";
        String signo = (numero > 0) ? "positivo" : (numero < 0) ? "negativo" : "cero";
        String rango = (numero > 0 && numero < 10) ? "un dígito" : 
                       (numero >= 10 && numero < 100) ? "dos dígitos" : "otros";
        
        System.out.println("\n=== OPERADOR TERNARIO ===");
        System.out.println("Número: " + numero);
        System.out.println("Es: " + tipo);
        System.out.println("Signo: " + signo);
        System.out.println("Rango: " + rango);
        
        // Ejemplo de ternario anidado
        String clasificacion = (numero >= 90) ? "Excelente" :
                               (numero >= 80) ? "Muy bueno" :
                               (numero >= 70) ? "Bueno" :
                               (numero >= 60) ? "Aceptable" : "Insuficiente";
        System.out.println("Clasificación (asumiendo que es una nota): " + clasificacion);
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * ================
 * Digite un número: 42
 * 
 * === OPERADOR TERNARIO ===
 * Número: 42
 * Es: par
 * Signo: positivo
 * Rango: dos dígitos
 * Clasificación (asumiendo que es una nota): Muy bueno
 */
